### ISSUE TITLE
Query returns 500 error

### ISSUE THREAD

**Issue by prohtex** [mstilkerich/carddavclient#27]
Hi there, thanks for your hard work on this client library.

I'm attempting to query a Baikal server for a single vcard based on matching email address field. This is what I have so far:

```
require "lib/carddav/vendor/autoload.php";

use MStilkerich\CardDavClient\{Account, AddressbookCollection, Config};
use MStilkerich\CardDavClient\Services\{Discovery};

Config::init($log, $httplog);
$account = new Account(<server>, ["username" => <username>, "password" => <pass>]);

$discover = new Discovery();
$abooks = $discover->discoverAddressbooks($account);

$abook = $abooks[0];

$contacts = $abook->query(
     ['EMAIL' => '/gmail/'],
     ['EMAIL'],
     false,
     1
);
```

What I'm experiencing is this: when the query returns no hits, that's fine. But when I enter something like "gmail" that I know will match many vcards, I get this error:

```
Exception: Expected Multistatus HTTP request was not successful (500 Internal Server Error): <?xml version="1.0" encoding="utf-8"?> <d:error xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns"> <s:sabredav-version>4.5.0</s:sabredav-version> <s:exception>Sabre\Xml\ParseException</s:exception> <s:message>The input element to parse is empty. Do not attempt to parse</s:message> </d:error> in /opt/www/harkawik.com/lib/carddav/vendor/mstilkerich/carddavclient/src/CardDavClient.php on line 491
```

I know the credentials are working fine, but am scratching my head at this error. Please help!

**Comment by mstilkerich** [comment]
Hi, do you use Digest authentication with Baïkal? (it's the default). Then try switching to Basic Authentication (in the Baikal admin interface) and see if it fixes your issue.

See [QUIRKS.md](https://github.com/mstilkerich/carddavclient/blob/master/doc/QUIRKS.md), "Internal server error on REPORTs with digest authentication" for details.

**Comment by mstilkerich** [comment]
Could you check if the latest master solves your issue?

**Comment by prohtex** [comment]
> Could you check if the latest master solves your issue?

It did, thank you

**Comment by mstilkerich** [comment]
I could reproduce that Baïkal would only check credentials when the query returns results, same as you experienced. Interestingly, with Nextcloud 28 using the same Sabre/DAV version authentication is required first.

I reported an issue sabre-io/dav#1548

**Comment by mstilkerich** [comment]
Fixed in 1.4.1
